### PR TITLE
Test/getVersionsOfAttribute and getSharedVersionsOfIdentityAttribute with peer shared attributes

### PR DESCRIPTION
**Pull request by Milena-Czierlinski**

### CODE REVIEW

File: src/useCases/consumption/attributes/GetVersionsOfAttribute.ts [hunk 1]
```diff
@@ -25,6 +25,11 @@ export class GetVersionsOfAttributeUseCase extends UseCase<GetVersionsOfAttribut
     }
 
     protected async executeInternal(request: GetVersionsOfAttributeRequest): Promise<Result<LocalAttributeDTO[]>> {
+        const attribute = await this.attributeController.getLocalAttribute(CoreId.from(request.attributeId));
+        if (typeof attribute === "undefined") {
+            throw RuntimeErrors.general.invalidPropertyValue("The given ID doesn't belong to a valid Attribute.");
```
Comment (slavistan) [inline src/useCases/consumption/attributes/GetVersionsOfAttribute.ts:30]: Nitpicking: The error message may suggest that the given ID then belongs to an _invalid_ attribute instead (it's not definitively clear to an outsider what the problem is). I politely suggest keeping it stupid and simple with `Attribute '${request.attributeId}' doesn't exist.`.

File: test/consumption/attributes-e2e.test.ts [hunk 2]
```diff
@@ -1061,13 +1061,13 @@ describe(SucceedRelationshipAttributeAndNotifyPeerUseCase.name, () => {
     });
 });
 
-describe("Get (shared) versions of identity attribute", () => {
+describe("Get (shared) versions of an identity attribute", () => {
```
Comment (slavistan) [inline test/consumption/attributes-e2e.test.ts:1064]: I suggest putting these tests into a dedicated `describe()` block, just as we did with the other use cases in that file. Makes it immediately apparent where which use cases is tested.

File: test/consumption/attributes-e2e.test.ts [hunk 3]
```diff
@@ -1061,13 +1061,13 @@ describe(SucceedRelationshipAttributeAndNotifyPeerUseCase.name, () => {
...
     let updatedRepositoryAttributeVersion1: LocalAttributeDTO;
     let updatedRepositoryAttributeVersion2: LocalAttributeDTO;
```
Comment (slavistan) [inline test/consumption/attributes-e2e.test.ts:1066]: Would love for these two to just be called `repositoryAttributeVersion{1,2}`, without the _updated_ prefix. The _updated_ part is an artifact from the implementation of the successions where we have two different states of the same attribute to maintain. You can hide that detail in the `beforeAll` test setup and not let it escape :-)

Changing the cumbersome name would make all the tests much easier to read. Same goes for the other `updated...` variable names outside of the test setup.

File: src/useCases/consumption/attributes/GetVersionsOfAttribute.ts [hunk 4]
```diff
@@ -25,6 +25,11 @@ export class GetVersionsOfAttributeUseCase extends UseCase<GetVersionsOfAttribut
     }
 
     protected async executeInternal(request: GetVersionsOfAttributeRequest): Promise<Result<LocalAttributeDTO[]>> {
+        const attribute = await this.attributeController.getLocalAttribute(CoreId.from(request.attributeId));
+        if (typeof attribute === "undefined") {
+            throw RuntimeErrors.general.invalidPropertyValue("Attribute '${request.attributeId}' doesn't exist.");
+        }
```
Comment (jkoenig134) [inline src/useCases/consumption/attributes/GetVersionsOfAttribute.ts:31]: record not found

**Comment by Milena-Czierlinski** [comment]
Mmh, the rebase did weird things and duplicated some commits.

**Comment by jkoenig134** [comment]
I think the PR title is a bit confusing..

1. this is no chore, you are actually changing code that is shipped
2. this feels like a bugfix, you are checking stuff that wasn't checked before but should be checked - please make sure to add this to the PR title and to the merge message

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

File: src/useCases/consumption/attributes/GetSharedVersionsOfIdentityAttribute.ts [hunk 5]
```diff
@@ -27,8 +27,12 @@ export class GetSharedVersionsOfIdentityAttributeUseCase extends UseCase<GetShar
     protected async executeInternal(request: GetSharedVersionsOfIdentityAttributeRequest): Promise<Result<LocalAttributeDTO[]>> {
         const attribute = await this.attributeController.getLocalAttribute(CoreId.from(request.attributeId));
 
-        if (!attribute || attribute.isRelationshipAttribute()) {
-            throw RuntimeErrors.general.invalidPropertyValue("The given ID doesn't belong to a valid IdentityAttribute.");
+        if (typeof attribute === "undefined") {
+            throw RuntimeErrors.general.recordNotFound(request.attributeId);
```
Comment (jkoenig134) [inline src/useCases/consumption/attributes/GetSharedVersionsOfIdentityAttribute.ts:31]:
```suggestion
            throw RuntimeErrors.general.recordNotFound(LocalAttribute);
```

File: src/useCases/consumption/attributes/GetVersionsOfAttribute.ts [hunk 6]
```diff
@@ -25,6 +25,11 @@ export class GetVersionsOfAttributeUseCase extends UseCase<GetVersionsOfAttribut
     }
 
     protected async executeInternal(request: GetVersionsOfAttributeRequest): Promise<Result<LocalAttributeDTO[]>> {
+        const attribute = await this.attributeController.getLocalAttribute(CoreId.from(request.attributeId));
+        if (typeof attribute === "undefined") {
+            throw RuntimeErrors.general.recordNotFound(request.attributeId);
```
Comment (jkoenig134) [inline src/useCases/consumption/attributes/GetVersionsOfAttribute.ts:30]: see above

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

**Review by jkoenig134 — APPROVED** [review]

**Comment by Milena-Czierlinski** [comment]
"The merge-base changed after approval." -> Looking this up, it seems I have to close and re-open this PR, in order to work around it.

**Pull request closed without merging**